Feat/profile editing

Tanrungthip opened the pull request:
Small Update from Team Suggestion
- Change the '8 chars' to '8 characters' in password field

Profile Editing Updated
- Name editing (first/last) with immediate UI updates
- Email update flow with secure verification
- Password change with reauthentication
- Full form validation and error handling
- Mobile-responsive UI with scrolling support

Issues Resolved
- Fixed scrolling issue (added ScrollView)
- Fixed name updates not displaying (added refreshUserData)
   : now its updated in both Journal/Profile page right after updated the name
- Improved email update clarity (separated current vs. new fields)
- Fixed email verification errors (switched to verifyBeforeUpdateEmail)
  : email updated link to Firebase authentication -> user need to verify the link in the email in order to complete the email update.


🙌🏻 Feel free to test the profile editing function, I might have miss something.

Tanrungthip committed:
feat(profile-editing): Add a new profile editing form UI

Tanrungthip committed:
fix(email-editing): fixed bug for updating email

annaxjgan commented on app/profile/edit.tsx:
```diff
@@ -0,0 +1,485 @@
+        [
+          {
+            text: 'OK',
+            onPress: () => router.back()
```
not sure if it's just me but this doesn't behave as expected for me, it directs me to the homepage instead :(. Maybe we can explicitly push to profile page?

Tanrungthip replied:
I actually made it redirect to the homepage, I forgot to think about the user flow. Thanks for catching that, it should redirect to the profile page.

annaxjgan commented on app/profile/edit.tsx:
```diff
@@ -0,0 +1,485 @@
+                  </TouchableOpacity>
+                </View>
+                {newPassword ? (
+                  <Text style={styles.helperText}>must contain 8 characters</Text>
```
nit: I missed out on this from my prev comment - but would adding 'at least' be less confusing - so "must contain at least 8 characters" ? it's abit misleading without it (not a big issue though)

Tanrungthip replied:
Thanks for pointing out, it's indeed sounds misleading, I will change it!

annaxjgan commented on app/profile/edit.tsx:
```diff
@@ -0,0 +1,485 @@
+      }
+
+      // Send email verification if new email is provided
+      let emailVerificationSent = false;
```
I got this error when trying to change the user's email, which I think is an extra security step enforced by firebase:
<img width="75" height="200" alt="IMG_0427" src="https://github.com/user-attachments/assets/787b1df9-48bd-4dcd-9b5a-a1bc231f5046" />

one suggestion could be to have a pop up modal prompting the user to re-enter their password 
```
const password = await promptUserForPassword(); // ← this could be a pop up alert modal
await reauthenticate(password);

// retry the action
await updateEmail(auth.currentUser!, newEmail);
```

annaxjgan commented on app/profile/edit.tsx:
```diff
@@ -0,0 +1,485 @@
+            <View style={styles.container}>
+            {/* Header with back button */}
+            <View style={styles.header}>
+              <TouchableOpacity onPress={() => router.back()} style={styles.backButton}>
```
same as above, this leads back to the homepage instead. Maybe do router.push()?

Tanrungthip committed:
fix(profile-editing): fix email changing bugs + back button redirection

Tanrungthip committed:
fix(safe-view-warning): SafeAreaView Deprecation Warning Fixed across all related files

Tanrungthip commented:
Pushed updates to address review comments:

Bug Fixes
- Fixed back button navigation
- Resolved email change authentication flow
- Updated SafeAreaView usage (removed deprecation warning)

Technical Changes
- Refined authentication logic
- Improved error handling and validation for email changing

Results
- Email changes now work as expected
- Navigation is consistent
- No warnings during runtime

annaxjgan commented:
I'm not able to login with my new email after changing it in the profile edit page and clicking on the verification link sent to my email.
Apart from this - all the other changes look good to me! 

Since this is not an urgent issue - I will merge this and we can create a separate bug fix task to look into this :)
